plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -627,20 +627,44 @@ type ServiceAffinity struct {
 	labels        []string
 }
 
-func NewServiceAffinityPredicate(podLister algorithm.PodLister, serviceLister algorithm.ServiceLister, nodeInfo NodeInfo, labels []string) algorithm.FitPredicate {
+// serviceAffinityPrecomputation should be run once by the scheduler before looping through the Predicate.  It is a helper function that
+// only should be referenced by NewServiceAffinityPredicate.
+func serviceAffinityPrecomputation(serviceLister algorithm.ServiceLister, podLister algorithm.PodLister) PredicateMetadataModifier {
```

Comment:
I really think that this should be a method on "ServiceAffinity" - in this case you won't need to pass any arguments to it.